types/web3.d.ts

```diff
@@ -0,0 +1,77 @@
+type Callback<T> = (err: Error | null, value: T) => void;
+type VoidCallback = (err: Error | null) => void;
+
+type Address = string;
+
+declare module 'web3' {
+  import { BigNumber } from 'bignumber.js';
+
+  type AnyNumber = number | string | BigNumber;
```

Comment:
`NumberLike` in `bignumber.js` is exaclty that

Reply:
I know that #33 depends on this PR, so don't bother fixing it in this branch, it should save you some headaches.

Reply:
`NumberLike` is defined in `@types/bignumber.js` which is not compatible with the newest `bignumber.js` package. The newest `bignumber.js` package ships with its own typings, unfortunately without `NumberLike` type.

Reply:
https://github.com/MikeMcl/bignumber.js/blob/master/bignumber.d.ts

Reply:
weird, good to know that